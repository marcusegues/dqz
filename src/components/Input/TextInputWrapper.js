// @flow
import React from 'react';
// $FlowFixMe
import { TextInput } from 'react-native';
import { pickerModalStyle } from '../Modals/styles/PickerModal';

type TextInputWrapperProps = {
  onChangeText: (text: string) => void,
  value: string,
  focus: boolean,
};

export class TextInputWrapper extends React.Component<TextInputWrapperProps> {
  componentDidMount() {
    if (this.props.focus) {
      this.textInput.focus();
    }
  }

  textInput: any;

  render() {
    const { onChangeText, value } = this.props;
    return (
      <TextInput
        keyboardType="numeric"
        maxLength={8}
        underlineColorAndroid="transparent"
        blurOnSubmit
        style={[pickerModalStyle.textInput, { alignSelf: 'center' }]}
        onChangeText={onChangeText}
        value={value}
        // eslint-disable-next-line no-return-assign
        ref={textInput => (this.textInput = textInput)}
      />
    );
  }
}
