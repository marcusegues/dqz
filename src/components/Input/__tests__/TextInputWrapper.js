// import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { TextInputWrapper } from '../TextInputWrapper';

// jest.mock('TextInputWrapper', () => {
//   return {
//     focus: jest.fn(),
//   };
// });

test('BlueInfoIcon component renders correctly', () => {
  const tree = renderer
    .create(<TextInputWrapper focus onChangeText={() => {}} value="44" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
//
// describe('TextInputWrapper', () => {
//   const component = renderer
//     .create(<TextInputWrapper onChangeText={() => {}} value="44" />)
//     .toJSON();
//   expect(component).toMatchSnapshot();
// });
