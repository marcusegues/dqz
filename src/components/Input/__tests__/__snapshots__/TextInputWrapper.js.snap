// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<TextInput
  allowFontScaling={true}
  blurOnSubmit={true}
  keyboardType="numeric"
  maxLength={8}
  onChangeText={[Function]}
  style={
    Array [
      Object {
        "borderBottomColor": "#9B9B9B",
        "borderBottomWidth": 1,
        "borderColor": "#A0A0A0",
        "color": "#1a1a1a",
        "fontFamily": "roboto_regular",
        "fontSize": 37.714285714285715,
        "height": 40,
        "textAlign": "center",
        "width": 100,
      },
      Object {
        "alignSelf": "center",
      },
    ]
  }
  underlineColorAndroid="transparent"
  value="44"
/>
`;

exports[`BlueInfoIcon component renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  blurOnSubmit={true}
  keyboardType="numeric"
  maxLength={8}
  onChangeText={[Function]}
  style={
    Array [
      Object {
        "borderBottomColor": "#9B9B9B",
        "borderBottomWidth": 1,
        "borderColor": "#A0A0A0",
        "color": "#1a1a1a",
        "fontFamily": "roboto_regular",
        "fontSize": 37.714285714285715,
        "height": 40,
        "textAlign": "center",
        "width": 100,
      },
      Object {
        "alignSelf": "center",
      },
    ]
  }
  underlineColorAndroid="transparent"
  value="44"
/>
`;

exports[`HeaderTitle renders according to snapshot 1`] = `
<TextInput
  allowFontScaling={true}
  blurOnSubmit={true}
  keyboardType="numeric"
  maxLength={8}
  onChangeText={[Function]}
  style={
    Array [
      Object {
        "borderBottomColor": "#9B9B9B",
        "borderBottomWidth": 1,
        "borderColor": "#A0A0A0",
        "color": "#1a1a1a",
        "fontFamily": "roboto_regular",
        "fontSize": 37.714285714285715,
        "height": 40,
        "textAlign": "center",
        "width": 100,
      },
      Object {
        "alignSelf": "center",
      },
    ]
  }
  underlineColorAndroid="transparent"
  value="44"
/>
`;
